scipy/fftpack/czt.py

```diff
@@ -0,0 +1,535 @@
+    y1 = zoomfft(x, [0, 2-2./len(y)])
+
+    # Check that zoomfft with oversampling is equivalent to zero padding
+    y2 = zoomfft(x, [0, 2-2./len(yover)], m=len(yover))
+
+    # Check that zoomfft works on a subrange
+    f1, f2 = w[3], w[6]
+    y3 = zoomfft(x, [f1, f2], m=3*over+1)
+    w3 = np.linspace(f1, f2, len(y3))
+    idx3 = slice(3*over, 6*over+1)
+
+    if not show:
+        plots = []
+    if plots != []:
+        import pylab
+    if 0 in plots:
+        pylab.figure(0)
+        pylab.plot(x)
+        pylab.ylabel('Intensity')
+    if 1 in plots:
+        pylab.figure(1)
+        pylab.subplot(311)
+        pylab.plot(w, abs(y), 'o', w, abs(y1))
+        pylab.legend(['fft', 'zoom'])
+        pylab.ylabel('Magnitude')
+        pylab.title('FFT equivalent')
+        pylab.subplot(312)
+        pylab.plot(w, np.angle(y), 'o', w, np.angle(y1))
+        pylab.legend(['fft', 'zoom'])
+        pylab.ylabel('Phase (radians)')
+        pylab.subplot(313)
+        pylab.plot(w, abs(y)-abs(y1))  # ,w, np.angle(y)-np.angle(y1))
+        # pylab.legend(['magnitude', 'phase'])
+        pylab.ylabel('Residuals')
+    if 2 in plots:
+        pylab.figure(2)
+        pylab.subplot(211)
+        pylab.plot(w, abs(y), 'o', wover, abs(y2), wover, abs(yover))
+        pylab.ylabel('Magnitude')
+        pylab.title('Oversampled FFT')
+        pylab.legend(['fft', 'zoom', 'pad'])
+        pylab.subplot(212)
+        pylab.plot(wover, abs(yover)-abs(y2),
+                   w, abs(y)-abs(y2[0::over]), 'o',
+                   w, abs(y)-abs(yover[0::over]), 'x')
+        pylab.legend(['pad-zoom', 'fft-zoom', 'fft-pad'])
+        pylab.ylabel('Residuals')
+    if 3 in plots:
+        pylab.figure(3)
+        ax1 = pylab.subplot(211)
+        pylab.plot(w, abs(y), 'o', w3, abs(y3), wover, abs(yover),
+                   w[3:7], abs(y3[::over]), 'x')
+        pylab.title('Zoomed FFT')
+        pylab.ylabel('Magnitude')
+        pylab.legend(['fft', 'zoom', 'pad'])
+        pylab.plot(w3, abs(y3), 'x')
+        ax1.set_xlim(f1, f2)
+        ax2 = pylab.subplot(212)
+        pylab.plot(wover[idx3], abs(yover[idx3])-abs(y3),
+                   w[3:7], abs(y[3:7])-abs(y3[::over]), 'o',
+                   w[3:7], abs(y[3:7])-abs(yover[3*over:6*over+1:over]), 'x')
+        pylab.legend(['pad-zoom', 'fft-zoom', 'fft-pad'])
+        ax2.set_xlim(f1, f2)
+        pylab.ylabel('Residuals')
+    if plots != []:
+        pylab.show()
+
+    err = norm(y-y1)/norm(y)
+    # print "direct err %g"%err
+    assert err < 1e-10, "error for direct transform is %g" % (err,)
+    err = norm(yover-y2)/norm(yover)
+    # print "over err %g"%err
+    assert err < 1e-10, "error for oversampling is %g" % (err,)
+    err = norm(yover[idx3]-y3)/norm(yover[idx3])
+    # print "range err %g"%err
+    assert err < 1e-10, "error for subrange is %g" % (err,)
+
+
+def _testscaled(x):
+    n = len(x)
+    norm = np.linalg.norm
+    assert norm(fft(x)-scaledfft(x)) < 1e-10
```

Comment:
Is this equivalent to `assert_allclose`, `assert_array_almost_equal_nulp` or `assert_array_max_ulp`?

    err = norm(y-y1)/norm(y)
    assert_(err < 1e-10, "error for direct transform is %g" % (err,))